packages/kit/src/runtime/load.js

```diff
@@ -5,9 +5,156 @@
 export function normalize(loaded) {
 	// TODO should this behaviour be dev-only?
 
-	if (loaded.error) {
-		const error = typeof loaded.error === 'string' ? new Error(loaded.error) : loaded.error;
+	if (
+		loaded.error ||
+		(loaded.status && loaded.status >= 400 && loaded.status <= 599 && !loaded.redirect)
+	) {
 		const status = loaded.status;
+		if (!loaded.error) {
+			let errorMessage = 'not assigned';
```

Comment:
There's no reason to initialize this to a string that's never used. I would just let it be undefined by default
```suggestion
			let errorMessage;
```